Ingress/controllers/nginx-third-party/ssl.go

```diff
@@ -0,0 +1,184 @@
+/*
```

Comment:
Since this file looks mostly like pure ssl, can we just move it up into controllers/ssl? this doesn't make sense if it has nginx specifics embedded in there